pkg/skaffold/build/cache/hash.go

```diff
@@ -38,15 +38,72 @@ import (
 
 // For testing
 var (
-	hashFunction           = cacheHasher
-	artifactConfigFunction = artifactConfig
+	newArtifactHasherFunc = newArtifactHasher
+	fileHasherFunc        = fileHasher
+	artifactConfigFunc    = artifactConfig
 )
 
-func getHashForArtifact(ctx context.Context, depLister DependencyLister, a *latest.Artifact, mode config.RunMode) (string, error) {
+type artifactHasher interface {
+	hash(ctx context.Context, a *latest.Artifact) (string, error)
+}
+
+type Artifacts map[string]*latest.Artifact
```

Comment:
You have a few places where you're `make(map[string]*latest.Artifact)` where this could be used instead.  Could be useful to have a creator function that takes an `[]latest.Artifact`?